### ISSUE TITLE
Possible bug when installing CGE using fppkg

### ISSUE THREAD

**Issue by Artq99** [castle-engine/castle-engine#158]
Hi,

today I tried to install CGE, but I didn't want to install Lazarus. I used CGE with pure FPC several years ago and I remember it worked.

The first problem I encountered was an error after running `fppkg install` in the main directory. During the compilation of the `CastleWindow` unit, the compiler could not find the library `JwaWinUser`. I checked this unit and it seems, that in the `fpmake.pp` file the dependency `winunits-jedi` (that is the provider for the unit) is missing. When I added the following line in the `fpmake.pp` file and run `fppkg install` once again, the installation process finished without errors:
```
if Defaults.OS in AllWindowsOSes then
    begin
      P.Dependencies.Add('winunits-base');
      P.Dependencies.Add('winunits-jedi'); // ADDED
      P.Dependencies.Add('fcl-registry');
    end;
```

The second problem was with the compilation of a program using the CGE. I tried to build the first sample app, the one that only opens a window, and it seemed that some units of the CGE have not been installed. And again, I added some lines to the `fpmake.pp` file, run `fppkg install` once again and finally I was able to compile and run the program. The missing lines:
```
P.Targets.AddUnit('castletransformextra.pas');
...
P.Targets.AddUnit('castleinternalbatchshapes.pas');
```

It seems to me that the `fpmake.pp` file has not been updated for some time and it can no longer install CGE out of the box. Please bear in mind, that I'm not a professional Pascal programmer, so there can be an another explanation to my problem.

Some details to my software:
OS: Windows 7 Pro 64x
FPC: 3.0.4. 32x
CGE: 6.5 32x

**Comment by michaliskambi** [comment]
Thanks for reporting! Yes, these both are definitely valid fixes. We introduced dependency on `winunits-jedi`recently, and we indeed added these 2 units without adding them to `fpmake.pp`.

We have Jenkins automatically testing fpmake compilation in https://github.com/castle-engine/castle-engine/blob/master/Jenkinsfile . But it didn't catch these 2 errors, because
- it tries fpmake on Linux
- it tries to compile engine with fpmake, but not `fppkg install` + compilation of some sample app.

So I'll extend this ticket a bit. I want to fix our Jenkinsfile to include testing this in the future, so that if I'll forget again to add a unit  to fpmake.pp -- Jenkins will notify me about this :)
